app/controllers/archive_controller.rb

```diff
@@ -0,0 +1,112 @@
+class ArchiveController < ApplicationController
+
+  def data_dictionary
+    fpm=Util::FilePresentationManager.new
+    @admin_schema_diagram=fpm.admin_schema_diagram
+    @schema_diagram=fpm.schema_archive_diagram
+    @data_dictionary=fpm.data_archive_dictionary
+    @table_dictionary=fpm.table_archive_dictionary
+    @tables = []
+    tabs=get_dictionary
+    header = tabs.first
+    (2..tabs.last_row).each do |i|
+      row = Hash[[header, tabs.row(i)].transpose]
+      if !row['table'].nil?
+        @tables << fix_attribs(row)
+      end
+    end
+
+    @view = []
+    view_tabs=get_views
+    view_header = view_tabs.first
+    (2..view_tabs.last_row).each do |i|
+      row = Hash[[header, view_tabs.row(i)].transpose]
+      if !row['table'].nil?
+        @view << fix_attribs(row)
+      end
+    end
+  end
+
+
+  def schema
+    set_diagrams_and_dictionaries
+    @show_dictionary_link = true
+    @project_schema_files=Share::Project.schema_diagram_file_names
+  end
+
+  def migration
+  end
+
+  def download
+  end
+
+  def snapshots
+    set_daily_monthly_snapshot_files
+  end
+
+  def pipe_files
+    set_daily_monthly_pipe_files
+  end
+
+  def covid_19
+    set_covid_pipe_files
+  end
+
+  private
+
+  def get_dictionary
+    Roo::Spreadsheet.open(Util::FileManager.new.table_archive_dictionary)
+  end
+
+  def get_views
```

Comment:
Is this being used at all?

Reply:
yes, it's being used in line 20 for parsing `aact_views.xlsx`